src/containers/rosterChecker/rosterChecker.jsx

```diff
@@ -0,0 +1,258 @@
+
+import { HotTable } from '@handsontable/react';
+import { Card, CardHeader, CardContent, TextField, Grid, Typography } from '@material-ui/core'
+import { withStyles } from '@material-ui/core/styles'
+
+import cloneDeep from 'lodash/cloneDeep';
+import isEmpty from 'lodash/isEmpty';
+import { withSnackbar } from 'notistack';
+
+import { setRealData } from 'actions/real/real'
+import { getArrangement } from 'services/arrangementService'
+
+const NAME_FIELD = "name";
+const VALIDITY_FIELD = "validity";
+const PRESENT = "present";
+const MISSING = "missing";
+const SHAMROCK_GREEN = "#0F9D58";
+const CINNABAR_RED = "#EA4335";
+
+const styles = theme => ({
+    sheet: {
+        marginLeft: 10,
+        marginTop: 10,
+        marginBottom: 10,
+        marginRight: 10,
+        borderStyle: "solid",
+        borderWidth: "1px",
+        borderColor: "#777"
+    },
+    card: {
+        background:"#fafafa"
+    },
+    cardHeader: {
+        paddingLeft: 10,
+        paddingTop: 10,
+        paddingBottom: 0,
+        paddingRight: 10
+    },
+    cardContent: {
+        height: "calc(100vh - 250px)",
+        overflow: "scroll"
+    }
+})
+
+class SheetView extends Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            arrangementId: "",
+            currentRoster: [],
+            newRoster: [],
+        }
+    }
+
+    generateColumnHeaders = () => {
+        return this.generateColumnDefs().map(col => col.data);
+    }
+
+    generateColumnDefs = () => {
+        return [
+            {
+                data: NAME_FIELD,
+                width: 220
+            },
+            {  
+                data: VALIDITY_FIELD,
+                width: 220,
+                readOnly: true,
+                renderer: this.renderValidity
+            }
+        ];
+    }
+
+    generateCurrentRoster = () => {
+        const itemList = []
+        console.log(this.props.real)
+        if (!isEmpty(this.props.real)) {
+            console.log("generating item list...");
+        
+            this.props.real.items.forEach(item => {
+                const validityOfItem = this.checkIfNameInRoster(item.name, this.state.newRoster) ? PRESENT : MISSING;
+                itemList.push({name: item.name, validity: validityOfItem});
+            });
+            this.setState({...this.state, currentRoster: itemList});
+        }
+        return itemList;
+    }
+
+    getItemFromItemName = (itemName) => {
+        return this.props.real.items.find(x => x.name === itemName);
+    }
+
+    renderValidity = (instance, td, row, col, prop, value, cellProperties) => {
+        if (value === PRESENT) {
+            td.style.backgroundColor = SHAMROCK_GREEN;
+        }
+        else if (value === MISSING) {
+            td.style.backgroundColor = CINNABAR_RED;
+        }
+        return td.innerHTML = value;
+    }
+
+    checkIfNameInRoster = (name, roster) => {
+        return roster.filter(row => {
```

Comment:
i think you can use `includes`

Reply:
i think includes doesn't take in a predicate